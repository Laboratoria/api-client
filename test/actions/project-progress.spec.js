import {
  addProjectProgress,
  getProjectProgress,
  updateProjectProgress,
} from '../../lib/actions/project-progress';

describe('getProjectProgress', () => {
  it('should be a function', () => {
    expect(typeof getProjectProgress).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(getProjectProgress({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});

describe('addProjectProgress', () => {
  it('should be a function', () => {
    expect(typeof addProjectProgress).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(addProjectProgress({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});

describe('updateProjectProgress', () => {
  it('should be a function', () => {
    expect(typeof updateProjectProgress).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(updateProjectProgress({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});
