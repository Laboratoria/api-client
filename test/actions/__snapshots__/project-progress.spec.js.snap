// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addProjectProgress should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-progress-add",
    "namespace": "default",
    "suffix": "PROJECT_PROGRESS_ADD",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_PROGRESS_ADD",
}
`;

exports[`getProjectProgress should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-progress",
    "namespace": "default",
    "suffix": "PROJECT_PROGRESS",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/cohorts/someone/projects/someone/progress",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_PROGRESS",
}
`;

exports[`updateProjectProgress should create an appropriate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-progress-update",
    "namespace": "default",
    "suffix": "PROJECT_PROGRESS_UPDATE",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/cohorts/someone/projects/someone/",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_PROGRESS_UPDATE",
}
`;
